rules.js

```diff
@@ -186,6 +186,103 @@ class sub_with_return extends rule(3) {
     }
 }
 
+class object_formatting extends rule(4) {
+    constructor() {
+        super()
+        this._node = 'object'
+        this.message = ``
+        this.spaces_after_colon = 1
+        this.spaces_before_colon = 0
+        this.trailing_commas = 'no' // 'no', 'required'
+        this.maxOneLine = 4
+        this.key_in_quotes = 'consistent' // 'allways', 'consistent', 'any'
+    }
+
+    check(node) {
+        let warnings = []
+        let quotedKeys = 0
+
+        const newlineCount = (token) => {
+            if (token.type != 'NL') return 0
+            return (token.value.match(/\n/g) || []).length
+        }
+        const spaceCount = (token) => {
+            if (token.type != 'ws') return 0
+            return token.value.length
+        }
+
+        const multiline = node.tokens.findIndex(item => newlineCount(item) > 0) > 0
+        const secondToken = node.tokens[1]
+        const secondToLast = node.tokens[node.tokens.length - 2]
+
+        if (multiline) {
+            if (newlineCount(secondToken) == 0) {
+                warnings.push(this.warning(secondToken, '`{` must be on its own line'))
+            }
+            if (newlineCount(secondToLast) == 0) {
+                warnings.push(this.warning(secondToLast, '`}` must be on new line'))
+            }
+        } else if (node.properties.length > 0) {
+            if (spaceCount(secondToken) != 1) {
+                warnings.push(this.warning(secondToken, '`{` must be followed by single space'))
+            }
+            if (spaceCount(secondToLast) != 1) {
+                warnings.push(this.warning(secondToLast, '`}` must be preceded by single space'))
+            }
+        }
+
+        if (multiline) {
+            let zip = node.tokens
```

Comment:
i don't really get what the name `zip` is supposed to mean

Reply:
changed to `pairs`